src/checkbox/checkbox.component.ts

```diff
@@ -126,26 +134,47 @@ export class Checkbox implements ControlValueAccessor, AfterViewInit {
 	/**
 	 * Sets the value attribute on the `input` element.
 	 */
-	@Input() value: string;
+	@Input() value: CheckboxValue;
 	/**
 	 * Used to set the `aria-label` attribute on the input element.
+	 *
+	 * @deprecated since v4 use the `ariaLabel` input instead
 	 */
 	// tslint:disable-next-line:no-input-rename
-	@Input("aria-label") ariaLabel = "";
+	@Input("aria-label") set ariaLabel(value: string) {
+		this._ariaLabel = value;
+	}
+
+	get ariaLabel() {
+		return this._ariaLabel;
+	}
+
+	// TODO: drop the `_`
+	// tslint:disable-next-line:no-input-rename
+	@Input("ariaLabel") _ariaLabel = "";
+
 	/**
 	 * Used to set the `aria-labelledby` attribute on the input element.
+	 *
+	 * @deprecated since v4 use the `ariaLabelledby` input instead
 	 */
 	// tslint:disable-next-line:no-input-rename
-	@Input("aria-labelledby") ariaLabelledby: string;
-	/**
-	 * Reflects whether the checkbox state is indeterminate.
-	 */
-	get indeterminate() {
-		return this._indeterminate;
+	@Input("aria-labelledby") set ariaLabelledby(value: string) {
+		this._ariaLabelledby = value;
+	}
+
+	get ariaLabelledby() {
+		return this._ariaLabelledby;
 	}
 
+	// TODO: drop the `_`
+	// tslint:disable-next-line:no-input-rename
+	@Input("ariaLabelledby") _ariaLabelledby: string;
```

Comment:
Does this need to be an input, or can it be a private/non-input property?

Reply:
I'm not sure what you mean ... Why would we want the label to not be an input?

Reply:
The setter above is an input remapped to `aria-label` ... this change maps this input to `ariaLabel`, but because we already have an `ariaLabel` property (the setter) we need to rename it to `_ariaLabel` ... We're deprecating `aria-label` and going with `ariaLabel`

Reply:
Oh okay I understand, thanks for clarifying